gemini/sample-apps/fashion-trends-prediction/app/data_prep/dataProcessing.py

```diff
@@ -0,0 +1,378 @@
+    """
+
+    if model == "Gemini":
+        return generateCaptionGemini(image_path)
+
+    elif model == "ImageQnA":
+        return generateCaptionImageQnA(image_path)
+
+
+def getPosts(user, previous, cnt=10, cookies={}, model="Gemini"):
+    """Gets a list of posts from an Instagram user.
+
+    Args:
+                    user (str): The username of the Instagram user.
+                    previous (list): A list of previous posts.
+                    cnt (int): The number of posts to get.
+                    cookies (dict): A dictionary of cookies.
+                    model (str): The model to use for generating the captions.
+
+    Returns:
+                    list: A list of posts.
+    """
+
+    org_cnt = cnt
+    userId = getId(user, cookies)
+    if userId is None:
+        return previous
+    params = {
+        "query_id": config["postid"],  # Fixed value for posts
+        "id": userId,  # User ID
+        "first": 12,
+    }
+
+    if len(previous) != 0:
+        latest_id = previous[0][0]
+    else:
+        latest_id = ""
+
+    print("latest id: ", latest_id, "\n")
+    posts = []
+
+    flag = False
+    while cnt > 0 and flag is False:
+        response = requests.get(
+            config["links"]["graphql"], params=params, cookies=cookies
+        )
+
+        print(response.status_code)
+
+        if response.status_code != 200:
+            print(response)
+            break
+
+        profile_data_json = response.text
+        parsed_data = json.loads(profile_data_json)
+
+        media = parsed_data["data"]["user"]["edge_owner_to_timeline_media"]
+
+        for i in range(len(media["edges"])):
+            if media["edges"][i]["node"]["__typename"] == "GraphVideo":
+                continue
+
+            postid = media["edges"][i]["node"]["id"]
+            postlink = media["edges"][i]["node"]["display_url"]
+            print("post id: ", postid)
+
+            if postid == latest_id:
+                flag = True
+                break
+
+            actual_img_path = "" + user + "$.png"
+            urllib.request.urlretrieve(postlink, actual_img_path)
+
+            try:
+                caption = generateCaption(actual_img_path, "Gemini")
+            except Exception as e:
+                print(e)
+            else:
+                posts = [(postid, postlink, caption)] + posts  # newest post stays first
+                cnt -= 1
+            finally:
+                os.remove(actual_img_path)
+
+            if cnt == 0:
+                break
+
+        params["after"] = media["page_info"]["end_cursor"]
+
+    posts = posts + previous
+    if len(posts) > org_cnt:
+        posts = posts[:org_cnt]
+
+    print("final cnt = ", cnt)
+    return posts
+
+
+def summarize_article(article_text: str) -> str:
+    """Summarizes an article.
+
+    Args:
+                    article_text (str): The article text.
+
+    Returns:
+                    str: The summary.
+    """
+
+    response = text_model.predict(
```

Comment:
This should probably be done with Gemini.

Reply:
Updated